api/clients/v2/disperser_client.go

```diff
@@ -213,9 +213,46 @@ func (c *disperserClient) DisperseBlob(
 		return nil, [32]byte{}, err
 	}
 
+	if verifyReceivedBlobKey(blobHeader, reply) != nil {
+		return nil, [32]byte{}, fmt.Errorf("verify received blob key: %w", err)
+	}
+
 	return &blobStatus, corev2.BlobKey(reply.GetBlobKey()), nil
 }
 
+// verifyReceivedBlobKey computes the BlobKey from the BlobHeader which was sent to the disperser, and compares it with
+// the BlobKey which was returned by the disperser in the DisperseBlobReply
+//
+// A successful verification guarantees that the disperser didn't make any modifications to the BlobHeader that it
+// received from this client.
+//
+// This function returns nil if the verification succeeds, and otherwise returns an error describing the failure
+func verifyReceivedBlobKey(
+	// the blob header which was constructed locally and sent to the disperser
+	blobHeader *corev2.BlobHeader,
+	// the reply received back from the disperser
+	disperserReply *disperser_rpc.DisperseBlobReply) error {
```

Comment:
```suggestion
	disperserReply *disperser_rpc.DisperseBlobReply,
) error {
```

Reply:
we should probably have a formatter to make the code standardized everywhere. I personally prefer this formatting, but I'm fine with whatever, just suggesting.

Reply:
[78c58a1d](https://github.com/Layr-Labs/eigenda/pull/1109/commits/78c58a1dfa1a3798fef730620a4eeabb8bbe16bf)

I don't have a strong preference. Strongly agree, this is the sort of thing that ought to be managed by a standard formatter.